### PR TITLE
Correct complexity calculator selection for scripts of Ride versions …

### PR DESCRIPTION
**Pull request by alexeykiselev**
…prior 5 added

### CODE REVIEW

File: pkg/ride/tree_evaluation_test.go [hunk 1]
```diff
@@ -2568,29 +2568,10 @@ func TestInvokeDAppFromDAppScript6(t *testing.T) {
...
-		Transfers:    make([]*proto.TransferScriptAction, 0),
-		Issues:       make([]*proto.IssueScriptAction, 0),
-		Reissues:     make([]*proto.ReissueScriptAction, 0),
-		Burns:        make([]*proto.BurnScriptAction, 0),
-		Sponsorships: make([]*proto.SponsorshipScriptAction, 0),
-		Leases:       make([]*proto.LeaseScriptAction, 0),
-		LeaseCancels: make([]*proto.LeaseCancelScriptAction, 0),
-	}
-	assert.Equal(t, expectedActionsResult, sr)
+	_, err = CallFunction(env, tree, "foo", proto.Arguments{})
```
Comment (esuwu) [inline pkg/ride/tree_evaluation_test.go:2571]: Why the function with a recursive call should return an error? I see you changed the number of invocation count, but how is that related with a recursive call?

Comment (alexeykiselev) [reply pkg/ride/tree_evaluation_test.go:2571]: Because it should return error in case of exceeding recursion limit

**Review by esuwu — APPROVED** [review]

File: pkg/ride/tree_evaluation_test.go [hunk 2]
```diff
@@ -2568,29 +2568,10 @@ func TestInvokeDAppFromDAppScript6(t *testing.T) {
...
+	require.Error(t, err)
```
Comment (nickeskov) [inline pkg/ride/tree_evaluation_test.go:2572]: Please, add assertion with error message to make sure that script has been failed with `invoke: too many internal invocations` reason

File: pkg/ride/tree_evaluation_test.go [hunk 3]
```diff
@@ -2760,29 +2740,10 @@ func TestReentrantInvokeDAppFromDAppScript6(t *testing.T) {
...
```
Comment (nickeskov) [inline pkg/ride/tree_evaluation_test.go:2744]: The same issue as above

File: pkg/state/script_caller.go [hunk 4]
```diff
@@ -207,7 +207,7 @@ func (a *scriptCaller) callAssetScript(tx proto.Transaction, assetID crypto.Dige
 }
 
 func (a *scriptCaller) invokeFunction(tree *ast.Tree, tx proto.Transaction, info *fallibleValidationParams, scriptAddress proto.WavesAddress) (ride.Result, error) {
-	env, err := ride.NewEnvironment(a.settings.AddressSchemeCharacter, a.state, a.settings.InternalInvokePaymentsValidationAfterHeight)
+	env, err := ride.NewEnvironment(a.settings.AddressSchemeCharacter, a.state, a.settings.InternalInvokePaymentsValidationAfterHeight, info.blockV5Activated, info.rideV5Activated)
```
Comment (nickeskov) [inline pkg/state/script_caller.go:210]: The last argument should be ` info.rideV6Activated`, not ` info.rideV5Activated`.

**Review by nickeskov — CHANGES_REQUESTED** [review]

**Review by nickeskov — APPROVED** [review]
